modules/tensor_mechanics/doc/content/documentation/systems/Materials/ComputeMultipleInelasticStress.md

```diff
@@ -1,14 +1,242 @@
+calculated inelastic strain increments for the other inelastic material models.
+The trial stress for each inelastic material model is computed from this initial
+elastic strain increment
+\begin{equation}
+  \label{eqn:trial_stress}
+  \sigma^{tr}_i = \begin{cases}
+                    C_{ijkl} \cdot \left( \epsilon^{el}_{old} + \Delta \epsilon^{el}_i \right) & \text{if $C_{ijkl}$ is isotropic}  \\
+                    \sigma_{old} + C_{ijkl} \cdot \left( \Delta \epsilon^{el}_i \right) & \text{if $C_{ijkl}$ is anisotropic}
+                  \end{cases}
+\end{equation}
+where $C_{ijkl}$ is the elasticity tensor for the material.
+
+The trial elastic strain increment and the trial stress are then passed to the
+$i^{th}$ inelastic material model, represented by the blue element [fig:multiple_materials].
+The inelastic model calculates the inelastic strain increment necessary to produce
+an admissible stress, as determined by the model's yield function. The specifics
+of the individual inelastic model admissible stress criteria are given in the
+documentation for the inelastic model. The elastic and inelastic strain increments,
+stress, and, optionally, the consistent tangentoperator are returned to
+`ComputeMultipleInelasticStress` from the $i^{th}$ inelastic material model.
+
+### Outer Iteration over Stress Difference
+
+After the inner iteration over all the inelastic material models is complete,
+the maximum and minimun stress values ($\sigma_{max}$ and $\sigma_{min}$) are
+identified from the set of stresses calculated by the individual inelastic models.
+An $L^2$ norm calculation is performed on the stress difference
+\begin{equation}
+  \label{eqn:l2_norm_stress}
+  L^2 \left(\Delta \sigma \right) = \sqrt{\Delta \sigma_{ij} \Delta \sigma_{ij}}
+    \text{, where } \Delta \sigma = \sigma_{max} - \sigma_{min}
+\end{equation}
+The $L^2$ norm of the stress difference is compared to the absolute and relative
+tolerances to determine if the solution from the combined inelastic material
+models is converged
+\begin{equation}
+  \label{eqn:l2_norm_convergence}
+  L^2 \left(\Delta \sigma \right) < \text{absolute tolerance, or }
+  \frac{L^2 \left(\Delta \sigma \right)}{L^2 \left(\Delta \sigma_o \right)} < \text{relative tolerance}
+\end{equation}
+where $L^2 \left(\Delta \sigma_o \right)$ is the $L^2$ norm from the very first
+outer iteration over all of the inelastic material models.
+The solution will not converge if the outer iteration loop, shown in the top half
+of [fig:multiple_materials], exceeds the maximum number of iterations set by the
+user.
+
+### Finalize Strains and the Jacobian Multiplier
+
+Once convergence on the stress is obtained, the method finalizes the calculation
+of the inelastic strains by applying a weighting factor, as shown in [fig:multiple_materials].
+This weighting factor has a default value of unity.
+
+The Jacobian multiplier, which is used in the [StressDivergenceTensors](Kernels/StressDivergenceTensors.md)
```

Comment:
The options available to the user are passed via `tangent_operator`, and are `elastic` where the tangent operator is never calculated, and the elasticity tensor is used instead, or `nonlinear` where the full tangent operator is calculated. How the method for `tangent_operator` calculation is set in the corresponding combined classes that inherit from `StressUpdateBase` (e.g. RadialReturnStressUpdate) and are `none`, in which the elastic tensor is used, `partial` in which is the partial math you have below, and `full` which is the nonlinear math you have below. In general, partial should be used for non-yielding models (creep, volumetric swelling) and full should be used for yielding models (plastic+creep). Hopefully this makes sense?

Reply:
How does this comment reconcile with the next comment down? e.g. if a partial tangent operator should be used for a creep model and a full tangent operator should be used with plasticity but the combination partial+ full isn't allowed, what is a user supposed to do?

Reply:
Re the consistent tangent operator.  Perhaps you could use material at http://mooseframework.org/wiki/PhysicsModules/TensorMechanics/CTO/   Perhaps you already have!  i haven't read this PR yet.

Reply:
Thank you @WilkAndy--that's lovely documentation!  I'll grab the general introduction you've written and use it here.